### PR TITLE
Cleanup DirectWrite Core support

### PR DESCRIPTION
**Pull request by riverar**
Made some changes per team feedback and added a functional sample.

[Ping @niklasb-ms.]

### CODE REVIEW

**Comment by niklasb-ms** [comment]
> Made some changes per team feedback and added a functional sample.
> 
> [Ping @niklasb-ms.]

Looks good to me! Thanks!

File: crates/samples/dwritecore/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            // Step 3.b: Get the font family names.
+            // Get the font family names by using the GetFamilyNames method. This method returns a
+            // IDWriteLocalizedStrings object that can point to multiple localized versions of the family name.
+
+            let family_names = family.GetFamilyNames()?;
+
+            // Step 3.c: Find the localized name.
+            // Get the font family name in the locale you want by using the FindLocaleName method.
+
+            let mut name_index = u32::default();
```
Comment (rylev) [inline crates/samples/dwritecore/src/main.rs:53]: nit: just doing `0` might be more clear?

File: crates/samples/dwritecore/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+            // Step 3.d: Get the length of the family name string
+            // Get the length of the family name string by using the GetStringLength method. Use this length
+            // to allocate a buffer large enough to hold the name and then get the font family name by using
+            // GetString.
+
+            let name_length = family_names.GetStringLength(name_index)?;
+
+            let mut name_buffer = Vec::<u16>::new();
+            name_buffer.resize((name_length + 1) as usize, 0);
```
Comment (rylev) [inline crates/samples/dwritecore/src/main.rs:66]:
```suggestion
            let mut name_buffer = vec![0u16; (name_length + 1) as usize];
```

Comment (riverar) [reply crates/samples/dwritecore/src/main.rs]: Thanks!

File: crates/samples/dwritecore/src/main.rs (hunk 1 above)
Comment (riverar) [reply crates/samples/dwritecore/src/main.rs:53]: Agreed.

**Pull request merged**